core/src/main/scala/quasar/destination/gbq/GBQFlow.scala

```diff
@@ -248,15 +254,71 @@ final class GBQFlow[F[_]: Concurrent](
               .as(DestinationError.invalidConfiguration(
                 (GBQDestinationModule.destinationType, 
                   config.sanitizedJson,
-                  ZNEList(s"Error creating GBQ job: ${resp.status.reason}"))).asLeft[Uri])
-
-            
+                  ZNEList(s"Error creating GBQ job: ${resp.status.reason}"))).asLeft[(Uri, GBQJob)])
         }
       }
     })
   }
 
-  private def upload(bytes: Stream[F, Byte], uploadLocation: F[Either[InitializationError[Json], Uri]]): Stream[F, Unit] = {
+  private def getJob(job: GBQJob): Stream[F, GBQJob] = {
+    val jobUrlString =
+      s"https://bigquery.googleapis.com/bigquery/v2/projects/${job.jobReference.projectId}/jobs/${job.id}"
+
+    val jobReq = credsF map { creds =>
+      Request[F](Method.GET, Uri.unsafeFromString(jobUrlString))
+        .withHeaders(creds)
+        .withContentType(`Content-Type`(MediaType.application.json))
+    }
+
+    Stream
```

Comment:
It doesn't seem to me like we need a `Stream` here; you need to get the body fully anyway to decode it, so we don't benefit from `Stream` and we can make it terser with `client.expectOr[GBQJob]`, and keeping it in just `F` (as opposed to `Stream[F`